Source/WebKit/UIProcess/Cocoa/_WKWarningView.mm

```diff
@@ -179,88 +180,58 @@
         });
     case WarningItem::GoBackButton:
         return WTF::switchOn(warningView.warning->data(), [&] (const WebKit::BrowsingWarning::SafeBrowsingWarningData&) {
             return narrow ? white : warning.tintColor;
         }, [&] (const WebKit::BrowsingWarning::HTTPSNavigationFailureData&) {
-            return white;
+            return narrow ? darkGray : warning.tintColor;
         });
     }
 #endif
     ASSERT_NOT_REACHED();
     return nil;
 }
 
-@interface WKSafeBrowsingExclamationPoint : ViewType
-@end
-
-@implementation WKSafeBrowsingExclamationPoint
-
-- (void)drawRect:(RectType)rect
+static ViewType* viewForIconImage(_WKWarningView *warningView)
 {
-    constexpr CGFloat centerX = exclamationPointSize / 2;
-    constexpr CGFloat pointCenterY = exclamationPointSize * 7 / 30;
-    constexpr CGFloat pointRadius = 2.25 * exclamationPointSize / 30;
-    constexpr CGFloat lineBottomCenterY = exclamationPointSize * 13 / 30;
-    constexpr CGFloat lineTopCenterY = exclamationPointSize * 23 / 30;
-    constexpr CGFloat lineRadius = 1.75 * exclamationPointSize / 30;
-    ViewType *warning = self.superview.superview;
+    NSString *symbolName;
+    WebCore::CocoaColor *color = colorForItem(WarningItem::WarningSymbol, warningView);
+    Boolean shouldSetTint = NO;
+    CGFloat imagePointSize = fontOfSize(WarningTextSize::Title).pointSize * imageIconPointSizeMultiplier;
+    WTF::switchOn(warningView.warning->data(), [&] (const WebKit::BrowsingWarning::SafeBrowsingWarningData&) {
+        symbolName = @"exclamationmark.circle.fill";
+    }, [&] (const WebKit::BrowsingWarning::HTTPSNavigationFailureData&) {
+        symbolName = @"lock.slash.fill";
+        shouldSetTint = YES;
+    });
 #if PLATFORM(MAC)
-    [colorForItem(WarningItem::ExclamationPoint, warning) set];
-    NSBezierPath *exclamationPoint = [NSBezierPath bezierPathWithOvalInRect:NSMakeRect(0, 0, exclamationPointSize, exclamationPointSize)];
-    [exclamationPoint appendBezierPathWithArcWithCenter: { centerX, lineBottomCenterY } radius:lineRadius startAngle:0 endAngle:180 clockwise:YES];
-    [exclamationPoint appendBezierPathWithArcWithCenter: { centerX, lineTopCenterY } radius:lineRadius startAngle:180 endAngle:360 clockwise:YES];
-    [exclamationPoint lineToPoint: { centerX + lineRadius, lineBottomCenterY }];
-    [exclamationPoint appendBezierPathWithArcWithCenter: { centerX, pointCenterY } radius:pointRadius startAngle:0 endAngle:180 clockwise:YES];
-    [exclamationPoint appendBezierPathWithArcWithCenter: { centerX, pointCenterY } radius:pointRadius startAngle:180 endAngle:360 clockwise:YES];
+    NSImageView *view = [NSImageView imageViewWithImage:[NSImage imageWithSystemSymbolName:symbolName accessibilityDescription:nil]];
+    view.symbolConfiguration = [NSImageSymbolConfiguration configurationWithPointSize:imagePointSize weight:NSFontWeightRegular scale:NSImageSymbolScaleLarge];
+    if (shouldSetTint)
+        view.contentTintColor = color;
```

Comment:
```suggestion
    RetainPtr view = [NSImageView imageViewWithImage:[NSImage imageWithSystemSymbolName:symbolName accessibilityDescription:nil]];
    [view setSymbolConfiguration:[NSImageSymbolConfiguration configurationWithPointSize:imagePointSize weight:NSFontWeightRegular scale:NSImageSymbolScaleLarge]];
    if (shouldSetTint)
        [view setContentTintColor:color];
```

(the current code is safe, but we are trying to use smart pointers where possible)